test/apps/iodemo/io_demo.cc

```diff
@@ -217,6 +227,54 @@ class P2pDemoCommon : public UcxContext {
         return conn->send_io_message(m->buffer(), opts().iomsg_size, m);
     }
 
+    bool send_recv_data_as_chunks(UcxConnection* conn, size_t data_size, uint32_t sn,
+                send_recv_data_t send_recv_data, UcxCallback* callback = EmptyCallback::get()) {
+        size_t left_size = data_size;
+        while (left_size > 0) {
+            size_t sz = (left_size >= opts().chunk_size) ? opts().chunk_size : left_size;
+            if (send_recv_data == SEND_DATA) {
+                conn->send_data(buffer(data_size - left_size), sz, sn, callback);
+            } else {
+                conn->recv_data(buffer(data_size - left_size), sz, sn, callback);
+            }
+            left_size -= sz;
+        }
+        return true;
+    }
+
+    bool send_data_as_chunks(UcxConnection* conn, size_t data_size, uint32_t sn,
+                                     UcxCallback* callback = EmptyCallback::get()) {
+        return send_recv_data_as_chunks(conn, data_size, sn, SEND_DATA, callback);
+    }
+
+    bool recv_data_as_chunks(UcxConnection* conn, size_t data_size, uint32_t sn,
+                                     UcxCallback* callback = EmptyCallback::get()) {
+        return send_recv_data_as_chunks(conn, data_size, sn, RECV_DATA, callback);
+    }
+
+    uint32_t get_chunk_cnt(size_t data_size) {
+        return (opts().chunk_size == 0) ? 1 : ((data_size + opts().chunk_size - 1)
```

Comment:
need to remove check for  (opts().chunk_size == 0)

Reply:
@yosefe I also considered as this. But SIZE_MAX/data_size always can be set/changed at run time,  however the default value of chunk size is set at the very beginning. It's inconvient to change the value of chunk size according to SIZE_MAX/data_size at run time. Prefer to make chunk size fixed during run time. Let me know please if you still don't agree this.

Reply:
Fixed